compiler/lib/deadcode.ml

```diff
@@ -242,15 +262,20 @@ let remove_empty_blocks ~live_vars (p : Code.program) : Code.program =
             (let branch = block.branch in
              match branch with
              | Branch cont -> Branch (resolve cont)
-             | Cond (x, cont1, cont2) -> Cond (x, resolve cont1, resolve cont2)
+             | Cond (x, cont1, cont2) ->
+                 let cont1' = resolve cont1 in
+                 let cont2' = resolve cont2 in
+                 if Code.cont_equal cont1' cont2'
```

Comment:
Maybe add a comment saying that this logic also exists in Eval.f or just remove it  from eval.
Note that his can introduce more deadcode if x is no longer used which means that Deadcode might not reach a fixpoint. This might be undesirable.

Reply:
I actually added this to reach a fixpoint faster,. If you have a complex pattern matching whose result is unused, it can take several iterations to reach a fixpoint: simplifying the conditional allows to empty the block, which allows to simplify a previous conditional. This happens when compiling `ocamlc`.

Maybe we can remove this logic from Eval.f, indeed.